hs_core/models.py

```diff
@@ -313,7 +313,9 @@ def page_permissions_page_processor(request, page):
     else:
         is_version_of = ''
 
-    permissions_allow_copy = request.user.uaccess.can_view_resource(cm)
+    permissions_allow_copy = False
+    if request.user.is_authenticated:
+        permissions_allow_copy = request.user.uaccess.can_view_resource(cm)
```

Comment:
can_view_resources will return False is the user is not authenticated or not able to view this resource, right? If that is the case, lines 316 and 317 are not really needed, right?

Reply:
OK. Never mind. I got it. Anonymous users need to be handled explicitly here.